src/main/java/uk/gov/hmcts/reform/prl/services/StmtOfServImplService.java

```diff
@@ -137,13 +142,63 @@ && isNotEmpty(caseData.getServiceOfApplication().getUnServedRespondentPack())) {
             }
             elementList.add(element(recipient));
         }
+
         caseData = caseData.toBuilder()
-            .stmtOfServiceAddRecipient(elementList)
+            .statementOfService(StatementOfService.builder()
+                                    .stmtOfServiceForApplication(appendStatementOfServiceToSoaTab(
+                                        caseData,
+                                        elementList
+                                    ))
+                                    .stmtOfServiceForOrder(appendStatementOfServiceToOrdersTab(
+                                        caseData,
+                                        elementList
+                                    ))
+                                    .stmtOfServiceAddRecipient(null)
+                                    .build())
             .build();
 
         return caseData;
     }
 
+    private List<Element<StmtOfServiceAddRecipient>> appendStatementOfServiceToSoaTab(
+        CaseData caseData,
+        List<Element<StmtOfServiceAddRecipient>> statementOfServiceListFromCurrentEvent) {
+
+        if (caseData.getStatementOfService()
+            .getStmtOfServiceWhatWasServed().equals(
+                StatementOfServiceWhatWasServed.statementOfServiceApplicationPack)) {
+            if (CollectionUtils.isNotEmpty(caseData.getStatementOfService().getStmtOfServiceForApplication())) {
+                statementOfServiceListFromCurrentEvent.addAll(caseData.getStatementOfService().getStmtOfServiceForApplication());
```

Comment:
Add all or add is enough?

Reply:
addAll is required as its a list of getStmtOfServiceForApplication